lib/results.go

```diff
@@ -32,6 +32,7 @@ type Result struct {
 	Body      []byte        `json:"body"`
 	Method    string        `json:"method"`
 	URL       string        `json:"url"`
+	Headers   string        `json:"headers"`
```

Comment:
I think it would be useful to marshal out headers in a structured format in the JSON encoding. Can we make this an http.Header with the intent?

Reply:
I thought about it but decide against it because:
* The marshaling of headers would be `map[string][]string`, which is not very easy to use
* You lose the ordering of headers that might be interesting for some type of debugging

Reply:
Could we then preserve the ordering with a different type? I'd like to be able to easily process results and their headers in the terminal with `jq` or similar.

How about:

```go
type Header struct { Key, Value string }

type Result struct {
    // ...
    Headers []Header
```

Reply:
How would you encode that in CSV format?

Reply:
I am not sure. We could just leave as you originally had it, unless you have a better idea :-)

Reply:
Not really. But then you want a JSON version for JSON and b64 wire format for CSV?

Reply:
Yeah, I think it’s totally fine for the different encodings to represent different fields differently.